express-post-json/server.ts

```diff
@@ -0,0 +1,44 @@
+import express from 'express';
+
+type Grade = {
+  id: number;
+  name: string;
+  course: string;
+  score: number;
+};
+
+type NextId = number;
+
+const grades: Record<number, Grade> = {};
+
+let nextId: NextId = 1;
+
+const app = express();
+
+app.get('/api/grades', (req, res) => {
+  const localArr: Grade[] = Object.values(grades);
+  res.json(localArr);
+});
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  const grade = req.body;
+  grade.id = nextId;
+  grades[nextId] = grade;
+  nextId++;
+
+  if (
+    typeof !grade.name === 'string' ||
+    typeof !grade.course === 'string' ||
+    typeof !grade.score === 'number'
+  ) {
+    res.status(204);
+  }
+  res.json(grade);
+  res.send();
```

Comment:
`res.json` calls `res.send()` for you, so you don't want to call `res.send`. Otherwise, you'll get an error that you're sending a second response.

Reply:
You still have both a call to `res.json()` and a call to `res.send()`. You can only have one. (Remember that `res.json()` _calls_ `res.send()`.